src/frontend/components/leftNav.jsx

```diff
@@ -6,30 +7,52 @@ import { BsFillBookmarkFill } from "react-icons/bs";
 import { FaListAlt } from "react-icons/fa";
 import { MdAccountCircle } from "react-icons/md";
 import { CgMoreO, CgMore } from "react-icons/cg";
+import { NavLink } from "react-router-dom";
+import { useAuth } from "frontend/context/authContext";
+import { useToast } from 'frontend/custom/useToast';
+import {useUser} from "frontend/context/userContext.jsx";
 
 const LeftNav = () => {
+    const [toggleLogoutButton, setToggleLogoutButton] = useState(false);
+    const { setIsLoggedIn } = useAuth();
+    const { showToast } = useToast();
+    const {loggedUserData} = useUser();
+
+    function navActive({ isActive }) {
+        return {
+            borderBottom: isActive ? "3px solid #A977FE" : "",
+            padding: isActive ? "0.5rem 1.5rem" : "",
+        }
+    }
+
+    const handleLogout = () => {
+        setIsLoggedIn(false);
+        showToast("Successfully Logged out", "success");
+    }
```

Comment:
I think this logging out should be responsibility be responsibility of `authContext` instead of some component.